set-based-validation/README.md

```diff
@@ -0,0 +1,28 @@
+# Set Based Consistency Validation
+A type of question that occurs a lot is "How can I check if an account with a particular email address already exists?". 
+In essence this signals the requirement to be able to validate a set, which is a more intricate process when it comes to applications which follow the CQRS paradigm. Without going in full detail here why this is a problem, a straightforward solution is to enhance the command model by adding a small look-up table which can be queried during such a validation step.
+
+There are several ways to invoke this set validation when using Axon, several of which are shared in this repository.
+
+Prior to checking if the email address already exists you need to save all the email addresses that are used. You can do this by using an event handling component:
+
+https://github.com/AxonIQ/code-samples/blob/set-validation/set-based-validation/src/main/java/io/axoniq/dev/samples/command/handler/AccountEventHandler.java
+
+You’ll need to make the `AccountEventHandler`'s processing group subscribing to be able to update the repository in the same thread that has applied the event; thus ensuring consistent updates of the set. You can achieve that by adding the following property to your `application.properties` file:
+
+axon.eventhandling.processors.emailEntity.mode=subscribing
+
+Subscribing processors will create and maintain a projection immediately after an event has been applied and thus are immediately consistent. Moreover lookup tables like this should always be owned by the command side _only_ and as such should not be exposed through a Query API on top of it.
+
+
+Now that you have a lookup table you can check if the email address exists before applying the event. We implemented this in three different ways:
+
+1. By using a command message dispatcher which does a check on the CreateAccountCommand: https://github.com/AxonIQ/code-samples/blob/set-validation/set-based-validation/src/main/java/io/axoniq/dev/samples/command/interceptor/AccountCreationDispatchInterceptor.java
```

Comment:
We can use 'beautiful' links using `[name](url)`, ex: [CreateAccountCommand](https://github.com/AxonIQ/code-samples/blob/set-validation/set-based-validation/src/main/java/io/axoniq/dev/samples/command/interceptor/AccountCreationDispatchInterceptor.java).
IMO they look better than the current one.

ps: same is true about the link, it won't work after we delete the current branch.